### ISSUE TITLE
UPSERT & UNIQUE indexes not getting along on distributed transactions

### ISSUE THREAD

**Issue by ghost** [orientechnologies/orientdb#4859]
I have set up a 3-node cluster (using both `2.0.13` and  `2.1.0`) and created the following class:

```
CREATE CLASS Person EXTENDS V
CREATE PROPERTY Person.uuid STRING
ALTER PROPERTY Person.uuid NOTNULL TRUE
ALTER PROPERTY Person.uuid MANDATORY TRUE
CREATE INDEX Person.uuid UNIQUE_HASH_INDEX
```

Now (in an over-simplification of my actual use case), trying to run the following batch against the server

```
script sql
BEGIN;
UPDATE Person SET uuid = '00000000-0000-0000-0000-000000000000' UPSERT WHERE uuid = '00000000-0000-0000-0000-000000000000';
UPDATE Person SET uuid = '00000000-0000-0000-0000-000000000000' UPSERT WHERE uuid = '00000000-0000-0000-0000-000000000000';
COMMIT;
end
```

yields

```
Error: com.orientechnologies.orient.core.exception.OTransactionException: Error on committing distributed transaction

Error: com.orientechnologies.orient.core.storage.ORecordDuplicatedException: Cannot index record #25:3: found duplicated key '00000000-0000-0000-0000-000000000000' in index 'Person.uuid' previously assigned to the record #25:2 RID=#25:2
```

Am I overlooking something here?

**Comment by lvca** [comment]
In the first update are you checking the `msisdn` attibute but then you insert `uuid`?

**Comment by bmcustodio** [comment]
@lvca I had changed the name of the field after opening the issue, for simplicity. Let's assume it is `uuid` (everywhere).

**Comment by lvca** [comment]
Why are you updating the same record in the same tx?

**Comment by lvca** [comment]
Fixed in v2.1.15
